### DISCUSSION TITLE
Help Needed Reading Boolean Input to D1 Mini

### DISCUSSION THREAD

**Discussion by Lasakro** [arendst/Tasmota#14170] [Q&A] [answered]
I have a D1 Mini configured in Tasmota as Generic 18 and D4 as an LED. D4 has a pulldown and the on board LED does switch state. How do I get that to show on the Tasmota main window so that I can read it in Tasmota?

**Comment by barbudor** [comment, marked as answer]
Configure a  "Relay" on D4 and you will be able to control the led from Tasmota

**Reply by Lasakro** [reply]
No LED to control. I need to read the input discrete value of D4. Tasmota should have an indicator of the state. Off or On. The D1 Mini on board LED changes but no software indication. D4 configured as either an LED or Relay gives the same results. Swich or Button don't seem to do anything with the Blue LED. All 4 don't provide any indication on the Tasmota page.

**Reply by barbudor** [reply]
For an input, you need to configure the GPIO as a "Switch". You will get state changes notifications in MQTT and can capture those changes with Rules

However there is no direct options in Tasmota to display the state of an input in the Web GUI.
It should be possible to do it using the Scripting-Language

**Reply by sfromis** [reply]
For a switch input, you can also use the command `Status 10` to query status, but of course, this does not become a fixture in the UI. It would be possible to use rules to control a "virtual relay" on an unused pin to have the "relay" being on/off in the UI.

**Reply by Lasakro** [reply]
I need to tread a furnace high temperature switch in a HA History chart as I make changes to the air flow. Is the D1 mini with Tasmota through MQTT not capable of such a simple task? I've been unsuccessfully searching for some digital input example but can't find one,

**Reply by sfromis** [reply]
As barbudor wrote, a digital input pin can be configured as a "Switch". Using SetOption114, Tasmota will transmit any state change via MQTT. You can also handle it locally using rules, and SwitchMode allows a lot of flexibility for how input state changes are handled. Obviously, you need to ensure that the electrical behavior matches the capabilities of ESP8266.

**Reply by barbudor** [reply]
> I need to tread a furnace high temperature switch in a HA History chart as I make changes to the air flow. Is the D1 mini with Tasmota through MQTT not capable of such a simple task? I've been unsuccessfully searching for some digital input example but can't find one,

Yes it can
Just forget about displaying it in the Web GUI of Tasmota especially if your goal is to get the status in HA

**Reply by Lasakro** [reply]
Helpful information, I thank you. I've set the D4 function to "Switch, disconnected all Switches from their respective Relays using SetOption114 1. My other setting are: SwitchMode1=1, SwitchTopic=0.

Still no success having a digital input Entity created in HA. I also tried this rule I found in the HA Buttons and Switches Wiki: Backlog Rule1 on Switch1#state do Publish stat/custom-topic/SWITCH1 %value% endon; Rule1

HA is seeing this controller status, but that's it.

**Reply by barbudor** [reply]
I'm not sure that Switches are supported by Tasmota/HA discovery
You probably have to create the HA configuration manually.
I suggest you join the [Tasmota Discord](https://discord.gg/vSVDFv7X) server and ask for help in the #home-automation channel.

**Reply by sfromis** [reply]
Take a look at "Binary Sensors" from the Tasmota HA docs.....
https://tasmota.github.io/docs/Home-Assistant/#binary-sensors

**Comment by Lasakro** [comment]
I gave up on Tasmota and found an immediate solution using ESPHome with GPIO Binary Sensor. I thank ones that pointed me in directions.
